src/adapters/WstEthChainlinkAdapter.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.21;
+
+import {IStEth} from "../interfaces/IStEth.sol";
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+
+import {ErrorsLib} from "../libraries/ErrorsLib.sol";
+
+/// @title WstEthChainlinkAdapter
+/// @author Morpho Labs
+/// @custom:contact security@morpho.org
+/// @notice wstETH/ETH exchange rate price feed.
+/// @dev This contract should only be used as price feed for `ChainlinkOracle`.
+contract WstEthChainlinkAdapter is AggregatorV3Interface {
+    uint8 public constant decimals = uint8(18);
+    string public constant description = "wstETH/ETH exchange rate";
+
+    IStEth public immutable ST_ETH;
+
+    constructor(address stEth) {
+        require(stEth != address(0), ErrorsLib.ZERO_ADDRESS);
+        ST_ETH = IStEth(stEth);
+    }
+
+    /// @notice Reverts as no Chainlink aggregator is used.
+    function version() external pure returns (uint256) {
+        revert();
+    }
+
+    /// @notice Reverts as it's not necessary for the ChainlinkOracle contract.
+    function getRoundData(uint80) external pure returns (uint80, int256, uint256, uint256, uint80) {
+        revert();
+    }
+
+    function latestRoundData() external view returns (uint80, int256, uint256, uint256, uint80) {
+        uint256 answer = ST_ETH.getPooledEthByShares(10 ** decimals);
+        require(answer <= uint256(type(int256).max), ErrorsLib.OVERFLOW);
```

Comment:
I agree but I don't like relying on `ChainlinkOracle`. What if we change the logic in `ChainlinkOracle`?

Reply:
The code has been audited multiple times, I don't think that it's a good idea. 

This wrapper is supposed to be used only by "ChainlinkOracle", so why is it a problem

Reply:
I don't have a strong opinion, @morpho-org/onchain-reviewers thoughts?

Reply:
Using `int256` does not revert when `WST_ETH.stEthPerToken() > type(int256).max` and silently overflows... I understand it will fail later in the oracle wrapper and that it is extremely unlikely to ever happen, but still this piece of software is weird if isolated: you can get a negative price from this adapter so we lose the invariant that the price is positive for example

and I expect this adapter to be used elsewhere in DeFi just the same way we started with Compound3's adapter

Reply:
> in this case let's revert again

I don't get the reasoning here